client/extensions/woocommerce/app/products/product-form-details-card.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import i18n from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import FormLabel from 'components/forms/form-label';
+import FormTextArea from 'components/forms/form-textarea';
+import FormTextInput from 'components/forms/form-text-input';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+
+export default class ProductFormDetailsCard extends Component {
+
+	static propTypes = {
+		product: PropTypes.shape( {
+			id: PropTypes.isRequired,
+			type: PropTypes.string.isRequired,
+			name: PropTypes.string,
+		} ),
+		editProduct: PropTypes.func.isRequired,
+	};
+
+	constructor( props ) {
+		super( props );
+
+		this.setName = this.setName.bind( this );
+		this.setDescription = this.setDescription.bind( this );
+		this.toggleFeatured = this.toggleFeatured.bind( this );
+	}
+
+	setName( e ) {
```

Comment:
We're going to have this pattern a lot, not just in this section but others. I highly recommend having an intermediate class that we could inherit some setter helper function from and then use it instead of writing separate settings for each field.

In the renderer then, you can pass the onChange handler the helper setter function with the setter field name, e.g.

`onChange={ this.setField( 'name' ) }`

We did something similar for /me using a form-base mixin, but I think that's fluxy and it doesn't understand WooCommerce edit state, so we will want to roll our own here.